builtin/providers/azurerm/config.go

```diff
@@ -58,6 +60,17 @@ func withRequestLogging() autorest.SendDecorator {
 	}
 }
 
+func setUserAgent(client *autorest.Client) {
+	var version string
+	if terraform.VersionPrerelease != "" {
+		version = fmt.Sprintf("%s-%s", terraform.Version, terraform.VersionPrerelease)
+	} else {
+		version = terraform.Version
+	}
+
+	client.UserAgent = fmt.Sprintf("HashiCorp Terraform v%s", version)
```

Comment:
Like the fact that we qualify the tag fully as 'HashiCorp Terraform'.  Not a big deal, but suggest changing the tag to "HashiCorp-Terraform-v%s" in order to avoid the blank spaces causing any issues with backend report filters.